jax/numpy/lax_numpy.py

```diff
@@ -2718,17 +2718,18 @@ def diagflat(v, k=0):
   return res
 
 
+@jit
+def _polyval(p, x):
+  shape = lax.broadcast_shapes(p.shape[1:], x.shape)
+  dtype = result_type(p, x)
+  y = lax.full_like(x, 0, shape=shape, dtype=dtype)
+  y, _ = lax.scan(lambda y, p: (y * x + p, None), y, p, unroll=64)
```

Comment:
This `unroll` number really seems to demand platform specific tuning.

On my laptop's CPU, `unroll=8` is about 30% faster than `unroll=1`, whereas `unroll=64` is 4x slower.

I haven't tested it on GPUs yet but given the cost of kernel launches suspect the conclusions would be pretty different.